### DISCUSSION TITLE
Help on DS1815+ with DSM7.1.1-42962 update 6

### DISCUSSION THREAD

**Discussion by richarda123** [007revad/Synology_HDD_db#245] [Q&A] [answered]
Hello,

Thanks a lot for making this script.
I was trying to expand a volume with a new HDD and I got the message about unsupported drive.
I found this script (thank you Reddit), downloaded the latest version (3.4.84) and ran it on my DS1815+.
It runs without any error but after rebooting, I still can't add the new HDD to expand the volume.

Unit: DS1815+
OS: DSM7.1.1-42962 update 6
Drive I'm trying to add: ST4000NM0033-9ZM170,EN09

Any help is welcome !

Richard

Below what I get when I run the script:

```
root@DS-Test:/volumeUSB3/usbshare/Synology_HDD_db-3.4.84# sudo -s ./syno_hdd_db.sh -nr
Synology_HDD_db v3.4.84
DS1815+-j DSM 7.1.1-42962-6 

Using model: ds1815+
Using options: -nr
Running from: /volumeUSB3/usbshare/Synology_HDD_db-3.4.84/syno_hdd_db.sh

HDD/SSD models found: 6
HN-M101MBB,2AR20004
ST4000NM0033-9ZM170,EN09
WD10JPVT-75A1YT0,01.01A01
WD20EARX-00PASB0,51.0AB51
WD40EFRX-68WT0N0,80.00A80
WD40EFRX-68WT0N0,82.00A82

No M.2 drives found

No M.2 PCIe cards found

No Expansion Units found

HN-M101MBB already exists in ds1815+_host_v7.db
HN-M101MBB already exists in ds1815+_host_v7.db.new
ST4000NM0033-9ZM170 already exists in ds1815+_host_v7.db
ST4000NM0033-9ZM170 already exists in ds1815+_host_v7.db.new
WD10JPVT-75A1YT0 already exists in ds1815+_host_v7.db
WD10JPVT-75A1YT0 already exists in ds1815+_host_v7.db.new
WD20EARX-00PASB0 already exists in ds1815+_host_v7.db
WD20EARX-00PASB0 already exists in ds1815+_host_v7.db.new
Updated WD40EFRX-68WT0N0 to ds1815+_host_v7.db
Updated WD40EFRX-68WT0N0 to ds1815+_host_v7.db.new
Updated WD40EFRX-68WT0N0 to ds1815+_host_v7.db
Updated WD40EFRX-68WT0N0 to ds1815+_host_v7.db.new

Support disk compatibility already enabled.

Max memory already set to 16 GB.

Drive db auto updates already disabled.

DSM successfully checked disk compatibility.

You may need to reboot the Synology to see the changes.

```

**Comment by 007revad** [comment]
From your comments on reddit it sounds like the drive database is corrupted, or DSM thinks it's corrupted. 
ST4000NM0033-9ZM170,EN09 and WD40EFRX-68WT0N0 both exist in the default ds1515+_host_v7.db file so they should always show as compatible.

Looking at the ds1815+_host_v7.db for DSM 7.1.1 I see it uses the old v7.db format. 

What does the following command return:
```
ls /var/lib/disk-compatibility | grep host
```

Can you attach a zipped copy of your `/var/lib/disk-compatibility/ds1815+_host_v7.db` file to your reply. You can copy it to your home folder with: 
```
cp /var/lib/disk-compatibility/ds1815+_host_v7.db $HOME
```

**Reply by richarda123** [reply]
Hello,

Thanks a lot for your help.

Result of the command: ls /var/lib/disk-compatibility | grep host

```
root@DS-Test:~# ls /var/lib/disk-compatibility | grep host
ds1815+_host_v7.db
ds1815+_host_v7.db.bak
ds1815+_host_v7.db.new
ds1815+_host_v7.db.new.bak
ds1815+_host_v7.release
ds1815+_host_v7.version
host_rule.db
root@DS-Test:~# 
```


And the file is attached:
[ds1815+_host_v7.db.zip](https://github.com/007revad/Synology_HDD_db/files/14320299/ds1815%2B_host_v7.db.zip)

I am starting getting worried. All my disks are now showing Health status : Not Supported in the Storage Manager. Because of that, I can't use any disk to repair or expand the storage pool. I can't either create any new storage pool with any of my disks.

Any Help is very welcome !

Thanks,

Richard

**Reply by richarda123** [reply]
And here is a copy of what I see in the Storage Manager (Health Status: Not Supported)

<img width="2087" alt="Screenshot 2024-02-17 at 23 12 15" src="https://github.com/007revad/Synology_HDD_db/assets/18413742/6e6dc57f-e9f0-4f79-b764-244dd4b628ea">

ALL Disks are impacted and even if I add a new Disk, it shows like this.... No way to repair, expand or whateved because of that Health Status.

**Reply by 007revad** [reply]
Okay, so the drives are healthy or uninitialized but their "health status" is unsupported. Interesting.

**Reply by richarda123** [reply]
And to be clear, it started BEFORE I launched your script.

Everything was working OK then, yesterday I had a message telling me that one of the Disk was failing.
The storage manager removed automatically the Disk from the Storage Pool and the Volume went to degraded status (no more redundancy). So I pull-out the failing Disk and replaced it by a new one. Then I launched the "Repair" process and everything worked smoothly.
Then, because I had another Disk at hand and I had an empty slot in my DS1815+, I decided to use the extra Disk to expand the volume. It is when I wanted to launch the Expand process that I got the error message telling me that I couldn't because the Health Status was not available.
Then I looked and I saw that all the disks were in this state.
It happened after I repaired the Volume, otherwise I couldn't have repaired it.
I found your script and used it to try to fix the problem but it doesn't seem to work.
So you're definitely not guilty of anything but if your script can help me getting out of this mess, I would be very grateful.

**Reply by 007revad** [reply]
> And to be clear, it started BEFORE I launched your script.

I did wonder about that.

Your db file is a newer version than the latest drive database update available for the DS1515+ (SynoOfflinePack-ds1815+-0020). It is as if it came from DSM 7.2.1. Did one of the drives come out of a NAS running 7.2.1?

**Reply by 007revad** [reply]
If you have a backup of your data and a DSM settings export you could try [Synology_DSM_reinstall](https://github.com/007revad/Synology_DSM_reinstall) using [7.1.1-42962 (with Update 1)](https://global.synologydownload.com/download/DSM/release/7.1.1/42962-1/DSM_DS1815%2B_42962.pat)

**Reply by 007revad** [reply]
Try the following commands:
```
sudo -i cp /tmp/synodrivedb.db /tmp/synodrivedb.db.orig
sudo -i cp /var.defaults/lib/smartmontools/synodrivedb.db /tmp/synodrivedb.db
```

**Reply by 007revad** [reply, marked as answer]
OK. Your issue was caused by Synology releasing a badly formatted smartmontools synodrive.db update. They have fixed it now so you need to go to "Storage Manager > HDD/SSD > Settings > Advanced" then scroll down to Drive Database and click on the Update Now button.

**Reply by richarda123** [reply]
Thanks. I just tried this and I got this strange behaviour:
<img width="1410" alt="Screenshot 2024-02-18 at 08 36 38" src="https://github.com/007revad/Synology_HDD_db/assets/18413742/27cd2b74-948e-44df-aa14-fe138d3970ce">

My station is indeed connected to the Internet. Anyway, I did the manual update process but the date of the last update remains the same: 2024-02-16 20:06:10.... and indeed this is probably when things started going down.

Also, when I do the manual process, I still get the same database version downloaded: SynoOfflinePack-ds1815+-0020. So maybe this is why it is not updating. Have you been able to get a more recent version ? (mine comes straight from the web site of Synology).
[SynoOfflinePack-ds1815+-0020.sa.zip](https://github.com/007revad/Synology_HDD_db/files/14321536/SynoOfflinePack-ds1815%2B-0020.sa.zip)

Is that file corrupted ?
If yes, could you please send me a good one ? (this one is the onmy one I can get from the Synology support section).

Thanks a lot for all the time you spend on my issue.

Richard

**Reply by 007revad** [reply]
The manual update (SynoOfflinePack) does not include the synodrive.db file. So you need to click on the Update Now button. The badly formatted update from the last 2 days is what causes the error message you're seeing.

Try the following commands to restore the original synodrive.db:
```
sudo -i
cp /var.defaults/lib/smartmontools/synodrivedb.db /tmp/synodrivedb.db
cp /var.defaults/lib/smartmontools/synodrivedb.db /var/lib/smartmontools/synodrivedb.db
```

**Reply by richarda123** [reply]
It Worked !!!!!
Thanks !
This is genius !
Is this solution ok in the long run ?

**Reply by 007revad** [reply]
You won't need to ever do that again... as long as Synology don't make the same mistake on the next smart drive database update.

**Reply by richarda123** [reply]
Well, thanks so much Sir !

Enjoy the rest of your day !
